src/js/components/AccountDropdownTrigger.js

```diff
@@ -0,0 +1,49 @@
+import mixin from "reactjs-mixin";
+import PropTypes from "prop-types";
+import React from "react";
+import { StoreMixin } from "mesosphere-shared-reactjs";
+
+class AccountDropdownTrigger extends mixin(StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.store_listeners = [
+      {
+        name: "metadata",
+        events: ["success"],
+        listenAlways: false
+      }
+    ];
+  }
+
+  componentDidUpdate() {
+    if (this.props.onUpdate) {
+      this.props.onUpdate();
+    }
+  }
+
+  render() {
+    const { primaryContent } = this.props;
+
+    return (
+      <a className="header-bar-dropdown" onClick={this.props.onTrigger}>
+        <span className="header-bar-dropdown-content text-overflow">
+          {primaryContent}
+        </span>
+      </a>
+    );
+  }
+}
+
+AccountDropdownTrigger.defaultProps = {
+  showCaret: false,
+  onTrigger() {}
+};
+
+AccountDropdownTrigger.propTypes = {
```

Comment:
Are we going to use those props later? right now it does not seem that they matter.

Should we also add here to this list `primaryContent` and `onTrigger`?

Reply:
👍